File: app/src/main/java/ru/argerd/repo/SplashActivity.java
package ru.argerd.repo;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;

import java.util.Timer;
import java.util.TimerTask;

public class SplashActivity extends AppCompatActivity {
    static {
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        Timer timer = new Timer();
        TimerTask timerTask = new TimerTask() {
            @Override
            public void run() {
                runOnUiThread(() -> startActivity(new Intent(SplashActivity.this, MainActivity.class)));
            }
        };
        timer.schedule(timerTask, 3000);
    }
}

File: app/src/main/java/ru/argerd/repo/SplashActivity.kt
package ru.argerd.repo

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import ru.argerd.repo.ui.AuthorizationActivity
import java.util.*

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        val timer = Timer()
        val timerTask: TimerTask = object : TimerTask() {
            override fun run() {
                runOnUiThread {
                    startActivity(Intent(this@SplashActivity,
                            AuthorizationActivity::class.java))
                }
            }
        }
        timer.schedule(timerTask, 3000)
    }
}